src/overlays/actors/ovl_En_Bat/z_en_bat.c

```diff
@@ -33,99 +40,522 @@ const ActorInit En_Bat_InitVars = {
+                (f32)0xFFFF * Rand_ZeroOne(), 0, BAD_BAT_PARAMS(this->switchFlag, this->paramFlags, 0), -1,
+                thisx->unk20, NULL);
+            BAD_BAT_GET_NUMBER_TO_SPAWN(thisx)--;
+        }
+    }
+}
+
+void EnBat_Destroy(Actor* thisx, PlayState* play) {
+    EnBat* this = THIS;
+
+    Collider_DestroySphere(play, &this->collider);
+}
+
+s32 EnBat_IsGraveyardOnSecondDay(PlayState* play) {
+    if ((CURRENT_DAY == 2) && (play->sceneNum == SCENE_BOTI)) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+void EnBat_StepAnimation(EnBat* this, s32 frameStep) {
+    s32 previousFrame = this->animationFrame;
+
+    this->animationFrame += frameStep;
+    if (this->animationFrame >= ARRAY_COUNT(sWingsDLs)) {
+        this->animationFrame -= ARRAY_COUNT(sWingsDLs);
+    }
+    if ((previousFrame < BAD_BAT_FLAP_FRAME) && (this->animationFrame >= BAD_BAT_FLAP_FRAME)) {
+        Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_FFLY_FLY);
+    }
+}
+
+void EnBat_SetupPerch(EnBat* this) {
+    this->collider.base.acFlags |= AC_ON;
+    this->collider.base.atFlags |= AT_ON;
+    this->collider.dim.worldSphere.center.x = this->actor.focus.pos.x;
+    this->collider.dim.worldSphere.center.y = this->actor.focus.pos.y;
+    this->collider.dim.worldSphere.center.z = this->actor.focus.pos.z;
+    this->actor.speedXZ = 0.0f;
+    this->actionFunc = EnBat_Perch;
+}
+
+void EnBat_Perch(EnBat* this, PlayState* play) {
+    EnBat_StepAnimation(this, 1);
+}
+
+void EnBat_SetupFlyIdle(EnBat* this) {
+    this->timer = 100;
+    this->collider.base.acFlags |= AC_ON;
+    this->actor.speedXZ = 3.5f;
+    this->actionFunc = EnBat_FlyIdle;
+}
+
+void EnBat_FlyIdle(EnBat* this, PlayState* play) {
+    s32 finishedRotStep;
+
+    EnBat_StepAnimation(this, 1);
+
+    finishedRotStep = Math_ScaledStepToS(&this->actor.shape.rot.y, this->yawTarget, 0x300);
+
+    if (this->actor.bgCheckFlags & 8) {
+        this->actor.bgCheckFlags &= ~8;
+        this->yawTarget = this->actor.wallYaw;
+    } else if (Math3D_XZDistanceSquared(this->actor.world.pos.x, this->actor.world.pos.z, this->actor.home.pos.x,
+                                        this->actor.home.pos.z) > 90000.0f) {
+        this->yawTarget = Actor_YawToPoint(&this->actor, &this->actor.home.pos);
+    } else if (finishedRotStep && (Rand_ZeroOne() < 0.015f)) {
+        this->yawTarget = this->actor.shape.rot.y +
+                          (((s32)((f32)0x1000 * Rand_ZeroOne()) + 0x1000) * ((Rand_ZeroOne() < 0.5f) ? -1 : 1));
+    }
+
+    finishedRotStep = Math_ScaledStepToS(&this->actor.shape.rot.x, this->pitchTarget, 0x100);
+
+    if ((this->actor.bgCheckFlags & 1) || (this->actor.depthInWater > -40.0f)) {
+        this->pitchTarget = -0x1000;
+    } else if (this->actor.world.pos.y < (this->actor.home.pos.y - 100.0f)) {
+        this->pitchTarget = -((s32)((f32)0x800 * Rand_ZeroOne()) + 0x800);
```

Comment:
```suggestion
        this->pitchTarget = -((s32)(0x800 * Rand_ZeroOne()) + 0x800);
```
It this specific cast needed?

Reply:
It's not, but I thought we preferred to be explicit about floats. There's quite a lot of others we can change like this. What do people think?